### PR TITLE
feat: Tag filter - API

### PR DESCRIPTION
**Pull request by tomthornton**
## `contentItemTags` Query
Screen Shot 2021-11-03 at 1 52 50 PM

### `getHeroListFeature`
```
query getHeroListFeature($featureId: ID!) {
    node(id: $featureId) {
      ...HeroListFeatureFragment
    }
}

fragment HeroListFeatureFragment on HeroListFeature {
    id
    title
    subtitle
    actions {
      id
      title
      subtitle
      action
      image {
        sources {
          uri
        }
      }
      relatedNode {
        ...RelatedFeatureNodeFragment
      }
    }
    primaryAction {
      title
      action
      relatedNode {
        ...RelatedFeatureNodeFragment
      }
    }
    heroCard {
      action
      title
      hasAction
      actionIcon
      labelText
      summary
      coverImage {
        sources {
          uri
        }
      }
      relatedNode {
        ...RelatedFeatureNodeFragment
      }
    }
  }

fragment RelatedFeatureNodeFragment on Node {
    id
    ... on Url {
      url
    }
    ... on ContentChannel {
      name
    }
    ... on Message {
      message
    }
  }
```

### Without tags
Feature Feed Query
Screen Shot 2021-11-03 at 1 56 50 PM

Hero List Feature Query
Screen Shot 2021-11-03 at 2 09 15 PM

Feature ID ```
HeroListFeature:f6b30a38eb547f1f75676ce0557a7e05bed66de665564bd0c5e37f0692676795b1b7ca4ba07c26064d4bad5f543b0cac297d6cb96b6c752eccd7ce3891b97a928c6c7ebff945821a6b7fbba69eec17585dad7f8a6b8c19716cdeedff993cfa837eb1eed5b7f256e4f955f8f4293cbe73389e502d544f8bdf9fa540c37f7aef493a88b0151bacfac54c8e8f72d6092c3d17ee625f97eea2dbcb45fdb9dd94d1d38e544488538c11679102f6c02f83d6c79156ca9dce3d526af43c6c8debb1e607ee20558abf011268d7f6ac3ca5fac4990d8bd458ba0fd96e8ab1938bb2a0d8b7549c18f476ea9b0e2c6d5445aebe8b081f507a18039d84bbd2d5e33350a945de34df717f255d961cc62e6bc00d913de8ef06d476cc7572998b67ce136de55fb7fd478e8ea73015952f4daf81d9c43e953c055c660757778dc42bd45c42075cba94fcbc80eb5c7d872eae9b88b6f5f99bfe21c8d22bdc3db62bd32468765db595f242ee2c2a2b049f9f882bd0f6879d1c```

### With tag

Feature Feed Query
Screen Shot 2021-11-03 at 1 57 12 PM

Hero List Feature Query
Screen Shot 2021-11-03 at 2 12 05 PM

Here, you can see that the `actions` have been filtered out and only the `heroCard` is returned because it is the only one that has that tag

Feature ID ```HeroListFeature:f6b30a38eb547f1f75676ce0557a7e05bed66de665564bd0c5e37f0692676795b1b7ca4ba07c26064d4bad5f543b0cac297d6cb96b6c752eccd7ce3891b97a928c6c7ebff945821a6b7fbba69eec17585dad7f8a6b8c19716cdeedff993cfa837eb1eed5b7f256e4f955f8f4293cbe73389e502d544f8bdf9fa540c37f7aef493a88b0151bacfac54c8e8f72d6092c3d17ee625f97eea2dbcb45fdb9dd94d1d38e544488538c11679102f6c02f83d6c79156ca9dce3d526af43c6c8debb1e607ee20558abf011268d7f6ac3ca5fac4990d8bd458ba0fd96e8ab1938bb2a0d8b7549c18f476ea9b0e2c6d5445aebe8b081f507a18039d84bbd2d5e33350a945de34df717f255d961cc62e6bc00d913de8ef06d476cc7572998b67ce136de55fb7fd478e8ea73015952f4daf81d9c43e953c055c660757778dc42bd45c42075cba94fcbc80eb5c7d872eae9b88b6f5f99bfe21c8d22bdc3db62bd32468765db59504b704741b322a4348b2afaa7b9d4ef3ac3ebfa6c45fd80744740642f12cf6c9```

### CODE REVIEW

**Comment by tomthornton** [comment]
@redreceipt need this one merged before I create the UI PR

File: packages/apollos-data-connector-postgres/src/action-algorithms/index.js [hunk 1]
```diff
@@ -217,13 +217,28 @@ class ActionAlgorithm extends PostgresDataSource {
...
 
     const items = await ContentItem.getFromCategoryIds(channelIds, {
       limit,
       skip,
       order: [['publishAt', 'DESC']],
+      include: [
+        {
+          model: this.sequelize.models.tag,
+          as: 'tags',
+          where: tags.length > 0 && {
```
Comment (redreceipt) [inline packages/apollos-data-connector-postgres/src/action-algorithms/index.js:236]: can you explain to me what this is doing?

Comment (tomthornton) [reply packages/apollos-data-connector-postgres/src/action-algorithms/index.js:236]: So basically what this does is it returns the all tags from every content item *if* tags is an empty array. I did this incase we want to use these tags later on the front-end. If tags is not empty, then the model is required and sequelize will only return content items that have at least one of the tags in the `tags` argument.

Comment (tomthornton) [reply packages/apollos-data-connector-postgres/src/action-algorithms/index.js:236]: So basically what this does is it returns the all tags from every content item if tags is an empty array. I did this incase we want to use these tags later on the front-end. If tags is not empty, then the model is required and sequelize will only return content items that have at least one of the tags in the tags argument.

**Comment by redreceipt** [comment]
I like it. Can you copy the same code over to the cluster in a matching PR?

**Comment by tomthornton** [comment]
@redreceipt cluster PR https://github.com/ApollosProject/apollos-cluster/pull/35

**Comment by tomthornton** [comment]
@redreceipt is this good to go or do I need to write tests for these?

**Review by redreceipt — APPROVED** [review]

**Pull request merged**